### PR TITLE
openthread: doc: describe there are multiple libraries for OpenThread

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Added info there are variants depending on stack and crypto.
Described Configuration used to prepeare each variant
Added info how to select stack variants

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

- [x] blocked by nrfconnect/sdk-nrf#2974

### CODE REVIEW

**Comment by MarekPorwisz** [comment]
Please have a look and let me know If there is something more needed. 
Note that I also removed settings related to ID/network name/channel/etc as those are fully configurable and shall not impact certification

File: openthread/README.rst [hunk 1]
```diff
@@ -8,64 +8,186 @@ This simplifies the OpenThread certification process of the final product.
...
+***************************
+
+The OpenThread libraries for Nordic devices vary depending on stack configuration and hardware cryptography capabilities.
+User is allowed to choose the stack configuration from one of the predefined values.
+Cryptograpy variant is selected automatically based on the microcontroller capabilities.
 
 Features of the pre-built libraries
 ***********************************
 
 The OpenThread pre-built libraries were created based on commit `656802a76acd2920b74ee4d49363f70aeb3b9621` of OpenThread,
```
Comment (edmont) [inline openthread/README.rst:21]: Commit hash should be updated, right?

File: openthread/README.rst [hunk 2]
```diff
@@ -8,64 +8,186 @@ This simplifies the OpenThread certification process of the final product.
...
+.. list-table::
+   :widths: 40 15 15 15
+   :header-rows: 1
 
+   * - OpenThread Kconfig option
+     - Master Image
+     - FTD Image
+     - MTD Image
+   * - :option:`CONFIG_OPENTHREAD_BORDER_AGENT`
+     - ``y``
```
Comment (edmont) [inline openthread/README.rst:35]: What about center-aligning these columns?

File: openthread/README.rst [hunk 3]
```diff
@@ -8,64 +8,186 @@ This simplifies the OpenThread certification process of the final product.
...
+     - ``n``
 
 Using the pre-built libraries
 *****************************
 
 Set the following Kconfig option to enable OpenThread pre-built libraries: :option:`CONFIG_OPENTHREAD_LIBRARY_1_1`.
 This option disables building OpenThread from source files and links pre-built libraries instead.
+
+To select OpenThread stack variant select one of the :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER` :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_FTD` :option:`CONFIG_OPENTHREAD_NORDIC_LIBRARY_MTD` options.
```
Comment (edmont) [inline openthread/README.rst:193]: Use commas to separate options or better a list.

**Review by edmont — COMMENTED** [review]
Wait for https://github.com/nrfconnect/sdk-nrfxlib/pull/268 before merging?

File: openthread/README.rst (hunk 2 above)
Comment (MarekPorwisz) [reply openthread/README.rst:35]: rst does not support alignment in tables.

File: openthread/README.rst (hunk 1 above)
Comment (MarekPorwisz) [reply openthread/README.rst:21]: I don't have the commit yet:) As you mentioned this will be updated after #268

**Comment by greg-fer** [comment]
Reviewed -- please check my commit. As mentioned to Marek and Lukasz, I don't believe we should mention how to use this libraries explicitly, but link to the related section in Configuring Thread in NCS.

Also, for me this PR is blocked by https://github.com/nrfconnect/sdk-nrf/pull/2974

**Review by greg-fer — APPROVED** [review]

File: openthread/README.rst [hunk 4]
```diff
@@ -3,69 +3,183 @@
...
-* :option:`CONFIG_OPENTHREAD_SLAAC` set to ``y``
-* :option:`CONFIG_OPENTHREAD_SNTP_CLIENT` set to ``y``
-* :option:`CONFIG_OPENTHREAD_SOURCES` set to ``y``
-* :option:`CONFIG_OPENTHREAD_THREAD_PRIORITY` set to ``8``
-* :option:`CONFIG_OPENTHREAD_THREAD_STACK_SIZE` set to ``6144``
-* :option:`CONFIG_OPENTHREAD_THREAD_VERSION` set to ``1.1``
-* :option:`CONFIG_OPENTHREAD_THREAD_VERSION_1_1` set to ``y``
-* :option:`CONFIG_OPENTHREAD_UDP_FORWARD` set to ``y``
-* :option:`CONFIG_OPENTHREAD_XPANID` set to ``de:ad:00:be:ef:00:ca:fe``
+The OpenThread pre-built libraries were created based on commit ``656802a76acd2920b74ee4d49363f70aeb3b9621`` of OpenThread,
```
Comment (lmaciejonczyk) [inline openthread/README.rst:20]:
```suggestion
The OpenThread pre-built libraries were created based on commit ``07f430dac6c400abdd358ba1173c0f94b94a5a80`` of OpenThread,
```

**Review by lmaciejonczyk — CHANGES_REQUESTED** [review]
The OpenThread commit differs

File: openthread/README.rst (hunk 4 above)
Comment (MarekPorwisz) [reply openthread/README.rst]: updated

**Review by lmaciejonczyk — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**